examples/ai/src/example-sessions.json

```diff
@@ -10,636 +10,324 @@
           "id": "hq9x8r95oennvhst8o6ollpm",
           "prompt": "draw time series",
           "streamMessage": {
-            "toolCallMessages": [
+            "parts": [
```

Comment:
@lixun910 would it be difficult to provide [`z.preprocess`](https://v3.zod.dev/?id=preprocess) to migrate the existing saved configs?

Reply:
it could looks smth like this, only the changes from `StreamMessageSchema` to `StreamMessagePartSchema` seem to be quite substantial:

```tsx
export const AnalysisResultSchema = z.object({
  id: z.string().cuid2(),
  prompt: z.string(),
  streamMessage: z.preprocess(
    (data) => {
      if (typeof data !== 'object' || data === null || !('parts' in data)) {
        const oldMessage = data as any;
        return {parts: oldMessage.toolCallMessages.map(msg => ({
          ...
        }))};
      }
      return data;
    },
    z.object({
      parts: z.array(StreamMessagePartSchema).optional(),
    }),
  ),
  errorMessage: ErrorMessageSchema.optional(),
  isCompleted: z.boolean(),
});
```